### ISSUE TITLE
(Accessibility) Implement alt text analogue for videos

### ISSUE THREAD

**Issue by qweertz** [LemmyNet/lemmy-ui#2779]
### Requirements

- [X] This is a feature request and not a bug report. Otherwise, please create a new [bug report](https://github.com/LemmyNet/lemmy-ui/issues/new?assignees=&labels=bug%2Ctriage&projects=&template=BUG_REPORT.yml) instead.
- [X] Please [check](https://github.com/LemmyNet/lemmy-ui/issues) to see if this request (or a similar one) already exists.
- [X] It's a single feature. Please don't request multiple features in one issue.

### Describe the feature you'd like

It would be nice to have the ability to add descriptive text to videos in a similar way to adding alt text to image posts. And for that text to be accessible to screen readers without cluttering the body of a post. This would bring all the benefits of the alt text function already present for image posts.

I am aware that, unlike `<img>`,  `<video>` does not have an `alt` attribute. However, from my understanding, the [`aria-label`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-label#associated_roles) attribute would do essentially the same thing.

---

Related question: If this were to be implemented, would it make sense to also add the separate ability to provide files in the WebVTT format (`.vtt`) to be used through the `<track>` tag?  This would enable ppl to:

>specify a timed text track (or time-based data) that can be displayed in parallel with the media element, for example to overlay subtitles or closed captions on top of a video or alongside audio tracks.
([taken from the MDN web docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/track))

This can be used to provide subtitles, captions, or chapters.

**Comment by travis-jeans** [comment]
In addition to the ability to upload a caption file for a video, the ability to add alternate audio tracks for audio description would be beneficial as well, for those who want to do it. On YouTube for example, there is no ability to upload audio description tracks which leads to creators making duplicate videos with alternate titles (one with default audio and one with audio description) which is not efficient and uses up unnecessary bandwidth. 
I have not used it myself but there is a "free for non profit use" video player I know of called [OzPlayer](https://www.accessibilityoz.com/ozplayer/) which has the ability to offer captions, transcripts, and audio description. Lemmy might fit in this category as an open source project. It might be worth asking them about it?
To find other such video players you can search for "video player with audio description" or "accessible video player" for different options in your search engine of choice.

**Comment by dessalines** [comment]
I think this is just a matter of showing the field for videos also, shouldn't be too bad.
